nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/TestMongoDBIndices.java

```diff
@@ -83,11 +84,13 @@ public void shouldFailWhenCreatingExistingChildNameDocument() {
             fail("should throw a ConcurrentUpdateException");
         } catch (ConcurrentUpdateException cue) {
             assertEquals(SC_CONFLICT, cue.getStatusCode());
+            String mongoDBName = Framework.getProperties().getProperty(MONGODB_DBNAME_PROPERTY);
             assertTrue(
                     cue.getMessage()
                        .contains(String.format(
                                "E11000 duplicate key error collection: %s.test index: ecm:parentId_1_ecm:name_1 dup key: { : \"%s\", : \"%s\" }",
-                               DEFAULT_MONGODB_DBNAME, folder.getId(), DOCUMENT_NAME)));
+                               mongoDBName, folder.getId(), DOCUMENT_NAME)));
+
```

Comment:
This is too complex, we don't care about the detailed format of the human-readable message that MongoDB returns. Just check for presence of `"E11000 duplicate key error"` and the folder id, and avoid all this complexity (which looks way too much like white-box testing).

Reply:
is the first devis what i did but i changed because most of times on our tests we check the whole message. but i will change  it to keep it more simple and readable :)